toolchain/check/handle_function.cpp

```diff
@@ -55,13 +56,110 @@ static auto BuildFunctionDecl(Context& context, bool is_definition)
       context.node_stack().PopIf<Parse::NodeKind::ImplicitParamList>().value_or(
           SemIR::InstBlockId::Empty);
   auto name_context = context.decl_name_stack().FinishName();
-  auto fn_node =
-      context.node_stack()
-          .PopForSoloParseNode<Parse::NodeKind::FunctionIntroducer>();
+  context.node_stack()
```

Comment:
This function had been about 100 lines, but with your changes it's now about 200. Please refactor the function to reduce size: typically around 100 lines should be an upper limit, and the style guide says to aim much lower.

https://google.github.io/styleguide/cppguide.html#Write_Short_Functions